src/main/java/training/supportbank/AccountCreate.java

```diff
@@ -0,0 +1,205 @@
+                    listOfNames.add(transactions[1]);
+                }
+
+                // Check if the 'To' names already exists in listOfNames, if not add to list of myAccounts & listOfNames
+                if (!listOfNames.contains(transactions[2])) {
+                    myAccounts.add(new Account(transactions[2]));
+                    listOfNames.add(transactions[2]);
+                }
+
+                BigDecimal cash;
+                try {
+                    cash = new BigDecimal(transactions[4]);
+                } catch (NumberFormatException nfe) {
+                    LOGGER.error("AMOUNT ERROR: Not a valid type for 'Amount': ERROR on line " + count + " of CSV file");
+                    continue;
+                }
+
+                myAccounts.stream()
+                        .filter(x -> x.getAccountName().equals(transactions[1]))
+                        .forEach(x -> x.addToBalance(cash));
+
+                myAccounts.stream()
+                        .filter(x -> x.getAccountName().equals(transactions[2]))
+                        .forEach(x -> x.subtractFromBalance(cash));
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        return myAccounts;
+
+    }
+
+    public static ArrayList<Account> readJSON(String path) {
+
+        ArrayList<Account> myAccounts = new ArrayList<>();
+        ArrayList<String> listOfNames = new ArrayList<>();
+
+        Gson gson = new Gson();
+
+        JsonObject[] array;
+
+        try  {
+            BufferedReader reader = new BufferedReader(new FileReader(path));
+
+            array = gson.fromJson(reader, JsonObject[].class);
+
+            for(JsonObject object: array) {
+
+                String from = object.get("fromAccount").toString().replace("\"", "");
+                String to = object.get("toAccount").toString().replace("\"", "");
+
+                if (!listOfNames.contains(from)) {
+                    myAccounts.add(new Account(from));
+                    listOfNames.add(from);
+                }
+
+                if (!listOfNames.contains(to)) {
+                    myAccounts.add(new Account(to));
+                    listOfNames.add(to);
+                }
+
+                BigDecimal cash;
+                try {
+                    cash = new BigDecimal(object.get("amount").toString());
+                } catch (NumberFormatException nfe) {
+                    //LOGGER.error("AMOUNT ERROR: Not a valid type for 'Amount': ERROR on line " + count + " of CSV file");
+                    continue;
+                }
+
+                myAccounts.stream()
+                        .filter(x -> x.getAccountName().equals(from))
+                        .forEach(x -> x.addToBalance(cash));
+
+                myAccounts.stream()
+                        .filter(x -> x.getAccountName().equals(to))
+                        .forEach(x -> x.subtractFromBalance(cash));
```

Comment:
Separating out the code here into two methods for JSON and CSV formats is definitely the right way to go.

Now that we have a working solution, we can start thinking about doing some refactoring to make the code as clean as possible.

Not a problem that you haven't done this, because we haven't covered these topics specifically, but having repeated code like this (lines 43-68) is a great chance to practise some refactoring.

Can you think of a way to change this slightly to move this logic into it's own method / static function? That way you can avoid writing these lines of code twice